examples/asr/jasper_eval.py

```diff
@@ -120,27 +122,25 @@ def main():
     )
     nemo.logging.info('================================')
 
-    (audio_signal_e1, a_sig_length_e1, transcript_e1, transcript_len_e1,) = data_layer()
+    # Define inference DAG
+    audio_signal_e1, a_sig_length_e1, transcript_e1, transcript_len_e1 = data_layer()
     processed_signal_e1, p_length_e1 = data_preprocessor(input_signal=audio_signal_e1, length=a_sig_length_e1)
     encoded_e1, encoded_len_e1 = jasper_encoder(audio_signal=processed_signal_e1, length=p_length_e1)
     log_probs_e1 = jasper_decoder(encoder_output=encoded_e1)
     predictions_e1 = greedy_decoder(log_probs=log_probs_e1)
 
-    eval_tensors = [
-        log_probs_e1,
-        predictions_e1,
-        transcript_e1,
-        transcript_len_e1,
-        encoded_len_e1,
-    ]
+    eval_tensors = [log_probs_e1, predictions_e1, transcript_e1, transcript_len_e1, encoded_len_e1]
 
-    evaluated_tensors = neural_factory.infer(tensors=eval_tensors, checkpoint_dir=load_dir, cache=True)
+    # inference
+    evaluated_tensors = neural_factory.infer(tensors=eval_tensors, checkpoint_dir=load_dir, cache=False)
```

Comment:
Any particular reason cache was set to False?

Reply:
Yes, I could not run inference after the refactoring because I run out of memory - my GPUs are only 16G and before I was using batch_size=32 but after the refactoring I couldn't run it even with a batch size of 8 - Same dataset as before. After turning cache to False, I could use batch_size of 32 again. I assume not everyone has GPUs with large memory so I assume this is a good default to have.

Reply:
That seems like a bug, let me try to push a fix first.

Reply:
Sounds good, let me know how I can help test.

Reply:
After a bit of discussion, we decided to have cache be False unless the lm_path is specified, in that case, we will default it to True. I'll address this in another PR.
The bug should be fixed in the current master, feel free to merge with master and see if caching works on your setup.